earth_enterprise/src/common/khCache.h

```diff
@@ -260,7 +305,9 @@ class khCache {
   void Prune(void) {
     CheckListInvariant();
     Item *item = tail;
-    while (item && (map.size() > targetMax)) {
+    while (item && (
+    ( (cacheObjectSizes > maxCacheMemory) && limitCacheMemory ) ||
+    ( (map.size() > targetMax) && !limitCacheMemory ) )) {
```

Comment:
I'm confused here - you're removing objects from the cache but I don't see you updating the cache size. I'm afraid this might be an infinite loop. It seems you would need to update the cache size every time you add or remove an object.

Reply:
cacheObjectSizes is updated whenever the deconstructor for MutableAssetHandleD or MutableDerivedAssetHandleD is called. I've tested to verify that when objects are added to the cache, the initial size of the object is calculated and cacheObjectSizes is then updated. Also, when objects are pruned, cacheObjectSizes is updated.

Reply:
Okay. Just ran some more specific tests and I see what you mean. Working to fix this now.